src/control/faults/faults.go

```diff
@@ -0,0 +1,113 @@
+package faults
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Resolution represents a potential fault resolution.
+type Resolution string
```

Comment:
What's the reason for this level of abstraction? The other fields are just strings.

Reply:
The thought was that it'd be good to have some predefined resolutions defined as consts and it's less maintenance to use those rather than doing string comparison if you need to check for the presence of one of those predefined resolutions. I'm not wedded to the idea, just seemed cleaner.

Reply:
Oh, maybe we got some lines crossed. I definitely agree with defining the common strings as consts. My concern was that creating a special Resolution type (vs. using the built-in string type) doesn't seem to add a lot, unless you have a vision for other methods to tack onto it. It's not clear to me why it's singled out to have its own type. But perhaps I am missing something here.

Reply:
Ah, yep. I'm ... not entirely sure what I was thinking there TBH. I agree that using a simple string is better for now.